### PR TITLE
Forecast: Prepare for translated forecasts

### PR DESCRIPTION
**Pull request by mrshu**
## Description of new Instant Answer, or changes
* Prepare for translated forecasts from DarkSky by passing the `$lang` parameter.

---

Instant Answer Page: https://duck.co/ia/view/forecast

### CODE REVIEW

**Comment by daxtheduck** [comment]
Metadata placeholder post
            
*If you haven't already, please edit your first comment to include a link to the IA page at the bottom.*
            
---
            
**How to find your IA page**
- New Instant Answer
  -- follow the steps [here](https://duck.co/ia/new_ia) to create your IA page.
              
- Existing Instant Answer
  -- find your IA page from this [list](https://duck.co/ia)

**Comment by moollaza** [comment]
@mrshu Thanks! This LGTM :+1:

**Comment by moollaza** [comment]
This is live on Beta:

https://beta.duckduckgo.com/?q=weather&kad=de_DE&ia=weather

File: lib/DDG/Spice/Forecast.pm [hunk 1]
```diff
@@ -22,7 +22,15 @@ handle remainder => sub {
     return if $_;
 
     my $loc_str = join " ", map { $loc->{$_} } @locs;
-    return $loc_str, 'current', {is_cached => 0};
+
+    # Default to English
+    my $parsed_lang = 'en';
+    # And try to extract language from locale
+    if ($lang && $lang->locale) {
+        ($parsed_lang) = $lang->locale =~ /_([A-Z]{2})$/;
```
Comment (bsstoner) [inline lib/DDG/Spice/Forecast.pm]: This isn't right, is it? I think locale is <lang>_<country>, e.g. en_US

Comment (bsstoner) [inline lib/DDG/Spice/Forecast.pm]: It's failing on all weather requests for me because it's sending lang to forecast as 'us'

**Pull request merged**